### PR TITLE
#104

### PR DESCRIPTION
**Pull request by packpaul**

### CODE REVIEW

File: celesta-core/src/main/resources/ru/curs/celesta/score/celesta.sql [hunk 1]
```diff
@@ -125,10 +125,12 @@ create table logsetup(
   constraint fk_logsetup_tables foreign key (grainid, tablename) references tables(grainid, tablename)
 );
 
+create sequence log_entryno;
```
Comment (inponomarev) [inline celesta-core/src/main/resources/ru/curs/celesta/score/celesta.sql:128]: We should not forget to increment system grain version (in `create grain...` statement). @ioanngolovko is now working on a branch with version 2.0, yours will be 2.01

File: celesta-core/src/main/resources/ru/curs/celesta/score/celesta.sql [hunk 2]
```diff
@@ -166,8 +168,10 @@ create table sequences(
   constraint fk_sequences_grains foreign key(grainid) references grains(id)
 ) with no version check;
 
+create sequence sessionlog_entryno;
+
 create table sessionlog (
-  entryno int identity not null primary key,
+  entryno int not null default nextval(sessionlog_entryno) primary key,
```
Comment (inponomarev) [inline celesta-core/src/main/resources/ru/curs/celesta/score/celesta.sql:175]: will you please add the following above this table as well:

```java
/**deprecated table -- to be removed*/
```

File: celesta-maven-plugin/src/main/java/ru/curs/celesta/plugin/CursorGenerator.java [hunk 3]
```diff
@@ -501,8 +501,7 @@ private static MethodSpec buildSetAutoIncrement(Map<String, ? extends ColumnMeta
         builder.addParameter(param);
 
         columns.entrySet().stream().filter(
-                e -> e.getValue() instanceof IntegerColumn && (((IntegerColumn) e.getValue()).isIdentity()
-                        || ((IntegerColumn)e.getValue()).getSequence() != null)
+                e -> ((e.getValue() instanceof IntegerColumn) && (((IntegerColumn) e.getValue()).getSequence() != null))
```
Comment (inponomarev) [inline celesta-maven-plugin/src/main/java/ru/curs/celesta/plugin/CursorGenerator.java]: maybe this should look more clear: 
```java
.filter(e -> e.getValue() instanceof IntegerColumn)
.filter(e -> ((IntegerColumn) e.getValue()).getSequence() != null)
.findAny()
```
(someone -- I am not pointing fingers  -- is always using `findFirst` where `findAny` suffice)

File: celesta-sql/src/main/java/ru/curs/celesta/dbutils/adaptors/DBAdaptor.java [hunk 4]
```diff
@@ -669,33 +665,6 @@ public String translateDate(String date) {
         return date;
     }
 
-    /**
```
Comment (inponomarev) [inline celesta-sql/src/main/java/ru/curs/celesta/dbutils/adaptors/DBAdaptor.java:672]: removed at last 🍾 🎆

File: celesta-sql/src/main/java/ru/curs/celesta/dbutils/adaptors/PostgresAdaptor.java [hunk 5]
```diff
@@ -167,11 +175,12 @@ public DbColumnInfo getColumnInfo(Connection conn, Column c) {
                             String tableName = c.getParentTable().getName();
 
                             if (sequenceName.equals(tableName + "_seq")) {
+                                // PP: probably dead code
```
Comment (inponomarev) [inline celesta-sql/src/main/java/ru/curs/celesta/dbutils/adaptors/PostgresAdaptor.java]: just remove the whole `if (sequenceName.equals(tableName + "_seq"))` thing

now we don't have any 'magic' sequence names and should treat all the sequences equally

File: celesta-sql/src/main/java/ru/curs/celesta/dbutils/meta/DbColumnInfo.java [hunk 6]
```diff
@@ -96,9 +87,7 @@ public boolean reflects(Column value) {
         }
 
         if (type == IntegerColumn.class) {
-            // Если свойство IDENTITY не совпадает -- не проверяем
-            if (isIdentity != ((IntegerColumn) value).isIdentity())
-                return false;
+            // Do nothing
```
Comment (inponomarev) [inline celesta-sql/src/main/java/ru/curs/celesta/dbutils/meta/DbColumnInfo.java]: just remove this `if` clause

File: celesta-sql/src/main/javacc/CelestaParser.jj [hunk 7]
```diff
@@ -24,18 +24,18 @@ import ru.curs.celesta.score.validator.IdentifierParser;
...
-                 try {
-                      return parser.sqlExpression(identifierParser);
-                 } catch (ParseException e) {
-                    throw new CelestaException("Invalid complex filter '%s': %s", filter, e.getMessage());
-                 }
-     } finally {
-         sr.close();
-     }
-   }
+        StringReader sr = new StringReader(filter);
```
Comment (inponomarev) [inline celesta-sql/src/main/javacc/CelestaParser.jj:27]: I see. We will definitely fix this try-with-resources issue in the next version of JavaCC!!!

File: celesta-sql/src/test/java/ru/curs/celesta/score/ParserTest.java [hunk 8]
```diff
@@ -294,7 +293,8 @@ public void test3() throws Exception {
     assertEquals(2, t.getForeignKeys().size());
   }
 
-  @Test
+  @Disabled("Recalculation of assert expected values is needed!")
```
Comment (inponomarev) [inline celesta-sql/src/test/java/ru/curs/celesta/score/ParserTest.java]: Please do not disable the whole test!! Just comment out the final checks, and replase them with
```java
assertTrue(g.getLength() > 10000);
assertTrue(g.getChecksum() <> 0);
```

File: celesta-test/src/test/java/ru/curs/celesta/dbutils/adaptors/AbstractAdaptorTest.java [hunk 9]
```diff
@@ -865,45 +856,31 @@ public void updateColumn2test() throws ParseException, IOException, SQLException
...
-        assertFalse(c.isIdentity());
         assertFalse(c.isNullable());
 
-        col.setNullableAndDefault(false, "identity");
+        col.setNullableAndDefault(false, "1");
         dba.updateColumn(conn, col, c);
-        dba.manageAutoIncrement(conn, t);
         c = dba.getColumnInfo(conn, col);
-        assertTrue(c.isIdentity());
         assertFalse(c.isNullable());
```
Comment (inponomarev) [inline celesta-test/src/test/java/ru/curs/celesta/dbutils/adaptors/AbstractAdaptorTest.java:885]: add assertion here that checks that default value for this field is "1"

File: celesta-test/src/test/java/ru/curs/celesta/dbutils/adaptors/AbstractAdaptorTest.java [hunk 10]
```diff
@@ -1169,18 +1141,19 @@ public void getFKInfo() throws ParseException, SQLException {
     @Test
     public void additionalCreateTableTest() throws ParseException {
         Grain g = score.getGrain(GRAIN_NAME);
+        SequenceElement t3s = g.getElement("aLongIdentityTableNxx_f1", SequenceElement.class);
         Table t3 = g.getElement("aLongIdentityTableNaaame", Table.class);
         try {
+            dba.createSequence(conn, t3s);
             dba.createTable(conn, t3);
             DbColumnInfo c = dba.getColumnInfo(conn, t3.getColumn("f1"));
-            assertTrue(c.isIdentity());
```
Comment (inponomarev) [inline celesta-test/src/test/java/ru/curs/celesta/dbutils/adaptors/AbstractAdaptorTest.java:1176]: add assertion here that checks that there is `default nextval` on the field

**Review by inponomarev — CHANGES_REQUESTED** [review]

**Review by inponomarev — APPROVED** [review]

File: celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java [hunk 11]
```diff
@@ -212,13 +212,13 @@ public final boolean tryInsert() {
                 loggingManager.log(this, Action.INSERT);
                 for (Column c : meta().getColumns().values())
```
Comment (ioanngolovko) [inline celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java:213]: Maybe 
```
meta().getColumns().values().stream()
    .filter(c -> c instanceOf IntegerColumn)
    .findFirst()
    .map(c -> (IntegerColumn)c)
    .filter(ic -> ic.getSequence() != null)
    .ifPresent(ic -> _setAutoIncrement(db().getCurrentIdent(conn(), meta())))

File: celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java [hunk 12]
```diff
@@ -633,19 +635,11 @@ boolean isEquivalentSpecific(BasicCursor bc) {
      * @param newValue
      *            значение, которое должно принять поле IDENITITY при следующей
      *            вставке.
+     * @deprecated
      */
     public final void resetIdentity(int newValue) {
```
Comment (ioanngolovko) [inline celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java]: Why not remove this method?

File: celesta-core/src/main/resources/ru/curs/celesta/score/celesta.sql [hunk 13]
```diff
@@ -34,7 +34,7 @@
  */
 
 /**Celesta system grain. Not for modification.*/
-create grain celesta version '1.14';
+create grain celesta version '2.01';
```
Comment (ioanngolovko) [inline celesta-core/src/main/resources/ru/curs/celesta/score/celesta.sql]: I  don't agree with this. Current version of `celesta` grain is 1.xx. So let it be as it is now, but with increment. We should not rely on other branches, that  haven't yet merged to master.

File: celesta-sql/src/test/java/ru/curs/celesta/score/ParserTest.java [hunk 14]
```diff
@@ -308,17 +306,20 @@ public void test4() throws Exception {
...
+    assertTrue(g.getChecksum() != 0);
 
     //TODO: non-stable values when using Git because of line ending conversion
     //hash sum calculation should be rewritten to be line-ending independent
+    //recalculation of assert expected values is needed
     if (";".equals(File.pathSeparator)){
       //Windows!
-      assertEquals(20767, g.getLength());
-      assertEquals(0x1754E6E7, g.getChecksum());
+//      assertEquals(20767, g.getLength());
```
Comment (ioanngolovko) [inline celesta-sql/src/test/java/ru/curs/celesta/score/ParserTest.java:338]: why?

**Review by ioanngolovko — DISMISSED** [review]
Maybe we need a bit more fixes

File: celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java (hunk 11 above)
Comment (packpaul) [reply celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java:213]: I don't see any issues with the current version - both versions are equivalent.

File: celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java (hunk 12 above)
Comment (packpaul) [reply celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java]: +1. Removed

File: celesta-core/src/main/resources/ru/curs/celesta/score/celesta.sql (hunk 13 above)
Comment (packpaul) [reply celesta-core/src/main/resources/ru/curs/celesta/score/celesta.sql]: Changed to 1.15

File: celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java (hunk 11 above)
Comment (ioanngolovko) [reply celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java:213]: The case without `break` is easier to read.

Comment (inponomarev) [reply celesta-core/src/main/java/ru/curs/celesta/dbutils/Cursor.java:213]: well, @ioanngolovko why convert everything to streams? I vote to leave it as is. I don't mind for..break cycle.

File: celesta-sql/src/test/java/ru/curs/celesta/score/ParserTest.java (hunk 14 above)
Comment (inponomarev) [reply celesta-sql/src/test/java/ru/curs/celesta/score/ParserTest.java:338]: The outcome of this test depends on local Git configuration. The file can be checked out both with Linux and Windows style EOLs, and sometimes the test fails.

Comment (ioanngolovko) [reply celesta-sql/src/test/java/ru/curs/celesta/score/ParserTest.java:338]: So...let's remove it!

Comment (packpaul) [reply celesta-sql/src/test/java/ru/curs/celesta/score/ParserTest.java:338]: @inponomarev So shall we remove the out-commented lines?

Comment (inponomarev) [reply celesta-sql/src/test/java/ru/curs/celesta/score/ParserTest.java:338]: do as you wish, I don't think this matters anything

**Pull request merged**